test/features/interceptor/authorization_interceptor_test.dart

```diff
@@ -0,0 +1,247 @@
+        const responseStatusCode = 401;
+
+        authorizationInterceptors.setTokenAndAuthorityOidc(
+          newToken: OIDCFixtures.tokenOidcWithTokenEmpty,
+          newConfig: OIDCFixtures.oidcConfiguration);
+
+        final result = authorizationInterceptors.validateToRefreshToken(responseStatusCode: responseStatusCode);
+
+        expect(result, false);
+      });
+
+      test('validateToRefreshToken should return false when condition `refreshToken is empty`', () async {
+        const responseStatusCode = 401;
+
+        authorizationInterceptors.setTokenAndAuthorityOidc(
+          newToken: OIDCFixtures.tokenOidcWithRefreshTokenEmpty,
+          newConfig: OIDCFixtures.oidcConfiguration);
+
+        final result = authorizationInterceptors.validateToRefreshToken(responseStatusCode: responseStatusCode);
+
+        expect(result, false);
+      });
+
+      test('validateToRefreshToken should return false when condition `Time not expired`', () async {
+        const responseStatusCode = 401;
+
+        final tokenOidc = TokenOIDC(
+          'dab123',
+          TokenId('dab123'),
+          'dab456',
+          expiredTime: DateTime.now().add(const Duration(days: 1))
+        );
+
+        authorizationInterceptors.setTokenAndAuthorityOidc(
+          newToken: tokenOidc,
+          newConfig: OIDCFixtures.oidcConfiguration);
+
+        final result = authorizationInterceptors.validateToRefreshToken(responseStatusCode: responseStatusCode);
+
+        expect(result, false);
+      });
+    });
+
+    group('QueuedInterceptorsWrapper test', () {
+      test('The request not should is not blocked when returning onError in case the token expires', () async {
+        final dioError401 = DioError(
+          error: {'message': 'Token Expired'},
+          requestOptions: RequestOptions(path: baseUrl),
+          response: Response(
+            statusCode: 401,
+            requestOptions: RequestOptions(path: baseUrl),
+          ),
+          type: DioErrorType.badResponse,
+        );
+
+        authorizationInterceptors.setTokenAndAuthorityOidc(
+          newToken: OIDCFixtures.tokenOidcExpired,
+          newConfig: OIDCFixtures.oidcConfiguration);
+
+        when(authenticationClient.refreshingTokensOIDC(
+            OIDCFixtures.oidcConfiguration.clientId,
+            OIDCFixtures.oidcConfiguration.redirectUrl,
+            OIDCFixtures.oidcConfiguration.discoveryUrl,
+            OIDCFixtures.oidcConfiguration.scopes,
+            OIDCFixtures.tokenOidcExpired.refreshToken
+        )).thenAnswer((_) async => OIDCFixtures.tokenOidc);
+
+        when(accountCacheManager.getCurrentAccount()).thenAnswer((_) async => AccountFixtures.aliceAccount);
+
+        dioAdapter
+          ..onPost(
+            baseUrl,
+            (request) => request.throws(401, dioError401),
+          )
+          ..onPost(
+            baseUrl,
+            (request) => request.reply(200, {'message': 'Retry request successfully!'})
+          );
```

Comment:
how about the case if on the next request still failed?

Reply:
please test with 2 more cases:
- case 1: both still dioError401
- case 2: first 401 - second 404

Reply:
I have written more for cases where there are multiple errors at the same time. Please see below.